src/components/_pages/certificates/detail/index.tsx

```diff
@@ -131,6 +135,8 @@ export default function CertificateDetail() {
         [isFetching, isDeleting, isUpdatingGroup, isUpdatingRaProfile, isUpdatingOwner, isRevoking, isRenewing, isRekeying],
     );
 
+    const [certificateNodes, certificateEdges] = useFlowChartValues(Resource.Certificates);
```

Comment:
is this hook needed? can't you just call `const { certificateNodes, certificateEdges } = transformCertifacetObjectToNodesAndEdges(certificate, users);`?

Reply:
Actually we are accessing the react redux state that is why we have created a hook since inside the hook we can access the store using useSelector..

Reply:
usually you already have those selectors, where you want to use flowchart, so you would just use them (instead of having creating hook, which is going to have lots of conditions and selectors inside of it...)

also please note, that those selectors are only selecting state, so if you haven't loaded it up before, it's not going to work...